proposals/exception-handling/Exceptions-formal-overview.md

```diff
@@ -0,0 +1,201 @@
+# 3rd proposal formal spec overview
+
+This is an overview of the 3rd proposal's formal spec additions, to aid in discussions concerning the proposed semantics.
+
+## Abstract Syntax
+
+### Types
+
+Tag Types
+
+```
+tagtype ::= [t*]→[]
+```
+
+### Instructions
+
+```
+instr ::= ... | throw x | rethrow l
+        | try bt instr* (catch x instr*)* (catch_all instr*)? end
+        | try bt instr* delegate l
+```
+
+### Modules
+
+Tags
+
+```
+tag ::= export* tag tagtype  | export* tag tagtype import
+```
+
+Modules
+
+```
+mod ::= module ... tag*
+```
+
+## Validation (Typing)
+
+#### Modification to labels
+
+To verify that a `try...delegate l` instruction refers to a label surrounding the instructions of a try block (call this a try-label), introduce a `kind` attribute to labels in the validation context, which is set to `try` when the label is a try-label.
+
+Similarly, to verify that the `rethrow l` instruction refers to a label surrounding the instructions of a catch block (call this a catch-label), we allow the `kind` attribute of labels in the validation context to be set to `catch` when the label is a catch-label.
+labelkind ::= try | catch
+labeltype ::= {result resulttype, kind labelkind?}
+C ::= {..., labels labeltype}
+The original notation `labels [t*]` is now an abbreviation for:
+
+```
+labels {result [t*], kind ε}
+```
+
+
+### Instructions
+
+
+```
+C.tags[x] = [t*]→[]
+-----------------------------
+C ⊢ throw x : [t1* t*]→[t2*]
+
+
+C.labels[l].kind = catch
+----------------------------
+C ⊢ rethrow l : [t1*]→[t2*]
+
+
+C.types[bt] = [t1*]→[t2*]
+C, labels {result [t2*], kind try} ⊢ instr* : [t1*]→[t2*]
+(C.tags[x] = [t*]→[] ∧
+ C, labels { result [t2*], kind catch } ⊢ instr* : [t*]→[t2*])*
+(C, labels { result [t2*], kind catch } ⊢ instr'* : []→[t2*])?
+-----------------------------------------------------------------------------
+C ⊢ try bt instr* (catch x instr'*)* (catch_all instr''*)? end : 	[t1*]→[t2*]
+
+
+C.types[bt] = [t1*]→[t2*]
+C, labels {result [t2*], kind try} ⊢ instr* : [t1*]→[t2*]
+C.labels[l].kind = try
+----------------------------------------------------------
+C ⊢ try bt instr* delegate l : [t1*]→[t2*]
+```
+
+## Execution (Reduction)
+
+### Runtime structure
+
+Stores
+
+```
+S ::= {..., tags taginst*}
+```
+
+Tag Instances
+
+```
+taginst ::= {type tagtype}
+```
+
+Module Instances
+
+```
+m ::= {..., tags a*}
```

Comment:
What is `a` here? Based on [this section](https://webassembly.github.io/spec/core/exec/runtime.html#module-instances), should we do something like
```
m ::= {..., tags tagtype*}
```
?

Reply:
The grammar should probably use `tagaddr` instead of `a`.

Reply:
In this case `a`means `tagaddr` (and `m` means `moduleinst`) because this is using the compact format of [this original formal-overview](https://github.com/WebAssembly/exception-handling/issues/87#issuecomment-517216137). The full core formal spec will have `tagaddr` and `moduleinst` there, as well as `blocktype` instead of `bt`, `tagidx` instead of `x`, etc. 

Would you prefer to use the longer format for the overview as well, @aheejin?

Reply:
@aheejin @rossberg : I'm not sure if you think I should use `tagaddr` instead of `a`, `blocktype` instead of `bt`, `labelidx` instead of `l`, etc, in this document as well, or only in the full spec document.

Reply:
In a _grammar definition_ you have to use the name of the productions, otherwise it would not really be a grammar. In other places, where you just pattern match syntactic phrases whose syntax is already fixed elsewhere, it typically is more convenient to use shorter names. It's a bit like the difference between a data type definition (grammar) where you use the type names and a pattern match (rule) where you have variable names.

Reply:
Just pushed a commit that addresses this in several places where grammar is defined.